tv_apps/smart-system/js/focus_manager.js

```diff
@@ -34,12 +36,19 @@
 
   proto.start = function ff_start() {
     window.addEventListener('screenchange', this);
-    this.startFallbackCatcher();
+    SettingsCache.get(SETTINGS_AUTO_FIXING, function(enabled) {
+      if (enabled !== false) {
+        this._autoFixingEnabled = true;
+        this.startFallbackCatcher();
+      }
+    }.bind(this));
   };
 
   proto.stop = function ff_stop() {
     window.removeEventListener('screenchange', this);
-    this.stopFallbackCatcher();
+    if (this._autoFixingEnabled) {
+      this.stopFallbackCatcher();
```

Comment:
I'd prefer resetting `this._autoFixingEnabled` to `false` to avoid exceptions if any.